cli/list/list.go

```diff
@@ -108,14 +108,26 @@ func parseBinaries(fileList []fs.DirEntry, programName string,
 					Str:   "master",
 				})
 			} else {
-				ver, err := version.Parse(versionStr)
-				if err != nil {
-					return binaryVersions, err
+				isRightFormat, _ := util.IsValidCommitHash(versionStr)
+
+				if isRightFormat {
+					ver := version.Version{}
+					if binActive == f.Name() {
+						ver.Str = versionStr + " [active]"
+					} else {
+						ver.Str = versionStr
+					}
+					binaryVersions = append(binaryVersions, ver)
+				} else {
+					ver, err := version.Parse(versionStr)
+					if err != nil {
+						return binaryVersions, err
+					}
+					if binActive == f.Name() {
+						ver.Str += " [active]"
+					}
+					binaryVersions = append(binaryVersions, ver)
```

Comment:
Please, add unit tests for this scenario.

Reply:
fixed